src/wallet/wallet2.cpp

```diff
@@ -2313,16 +2311,36 @@ void wallet2::load(const std::string& wallet_, const std::string& password)
 
       std::stringstream iss;
       iss << cache_data;
-      boost::archive::binary_iarchive ar(iss);
+      cereal::PortableBinaryInputArchive ar(iss);
       ar >> *this;
+      if (m_account_public_address.m_spend_public_key != m_account.get_keys().m_account_address.m_spend_public_key ||
```

Comment:
I do not understand the logic behind this check. `m_account` does not appear to be loaded from file, did I miss this? And if it was, is the goal to see if the same value differs in another location (i.e. the key is stored twice in the file)?